<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>893</width>
    <height>601</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Зашифровать/Расшифровать</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="3" column="0">
         <widget class="QFrame" name="keyFrame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Ключ:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="genKey">
             <property name="text">
              <string>Сгенерировать</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="fromFile">
             <property name="text">
              <string>Из файла</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="alphBox">
             <item>
              <property name="text">
               <string>Eng</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Rus</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="handmadeKey">
             <property name="text">
              <string>Ввести вручную</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout">
             <item>
              <widget class="QLineEdit" name="keyLine"/>
             </item>
             <item>
              <widget class="QLineEdit" name="key2Line"/>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPushButton" name="keyPair">
             <property name="text">
              <string>Ключ-значение</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="rsaKeyButton">
             <property name="text">
              <string>RSA-ключ</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="cardanButton">
             <property name="text">
              <string>Кардано-ключ</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QComboBox" name="cyphersBox">
          <item>
           <property name="text">
            <string>Простая замена</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Сдвиговый шифр замены</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Гаммирование</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Одиночная перестановка</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Двойная перестановка</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>RSA</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Маршрутная перестановка</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Полиалфавитная замена</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="4" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Открытый текст:</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QCheckBox" name="plainHex">
            <property name="text">
             <string>hex</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="openPlainText">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="clearPlainTextButton">
            <property name="text">
             <string>X</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="7" column="0">
         <widget class="QPlainTextEdit" name="cypherText"/>
        </item>
        <item row="2" column="0">
         <widget class="QFrame" name="modeFrame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QRadioButton" name="encrypt">
             <property name="text">
              <string>Зашифровать</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="decrypt">
             <property name="text">
              <string>Расшифровать</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QPlainTextEdit" name="plainText"/>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Шифр</string>
          </property>
         </widget>
        </item>
        <item row="8" column="0">
         <widget class="QPushButton" name="doButton">
          <property name="text">
           <string>Выполнить</string>
          </property>
         </widget>
        </item>
        <item row="6" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Шифротекст:</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QCheckBox" name="cryptHex">
            <property name="text">
             <string>hex</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="openCyphText">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="clearCyphTextButton">
            <property name="text">
             <string>X</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Криптоанализ</string>
       </attribute>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
